xoinvader/background.py

```diff
@@ -0,0 +1,261 @@
+
+        :param line: line to add
+        :type line: string
+        """
+        self.lines.append(line)
+        self.length += 1
+
+    def __getitem__(self, index):
+        return self.lines[index]
+
+    def __setitem__(self, index, item):
+        self.lines[index] = item
+
+
+def load_chunks(filename, trim_width=None):
+    """Load chunks from file.
+
+    File format:
+    ```
+    -~chunk~- <chunk_name>
+    <data>
+    ...
+    ```
+
+    `~chunk~` is magic string, and chunk spans until next magic line, or
+    until end of file.
+
+    Completely blank lines are ignored.
+    If `trim_width` is provided, all lines are truncated to this length.
+
+    :return: list of loaded chunks
+    :rtype: list
+
+    :param filename: path to file to load chunks from
+    :type filename: string
+
+    :param trim_width: width to truncate long lines to
+    :type trim_width: integer"""
+
+    with open(filename, "r") as f:
+        chunks = []
+        current_chunk = None
+        names = []
+        for line in f:
+            line = line.rstrip('\n')
+            if not line:
+                continue
+            elif line.startswith(Chunk.__MAGIC__):
+                name_split = line.split()
+                if len(name_split) < 2:
+                    raise ValueError("Chunks with empty names are not allowed")
+                name = name_split[1].strip()
+                if name in names:
+                    raise ValueError("Name {} already defined in background "
+                                     "file {}".format(name, filename))
+                current_chunk = Chunk(name)
+                chunks.append(current_chunk)
+                names.append(name)
+                continue
+            elif not current_chunk:
+                continue
+            # ok, now we got both chunk and something to store in it
+            if trim_width:
+                line_add = line[:trim_width]
+            else:
+                line_add = line
+            current_chunk.add_line(line_add)
+        if not chunks:
+            raise ValueError("File {} does not contain "
+                             "any chunks".format(filename))
+        return chunks
+
+
+class Background(Renderable):
+    """Class for storing and managing level background.
+    Manages background rendering and changing in time.
+
+    :param loop: flag if current chunk should loop after it had ended
+    :type loop: boolean
+
+    :param loop_all: flag if whole background should repeat from beginning
+    :type loop: boolean
+
+    :param speed: speed of background advance. Can be negative to represent
+    moving backwards
+    :type speed: integer
+
+    :param chunks: list of background chunks
+    :type chunks: list
+
+    :param background: list of background lines. You may change it at any time.
+    This list directly converts into Surface which then goes to the renderer
+    :type list:
+
+    """
+    render_priority = -1000
```

Comment:
Comment please with "TODO: render-priority" that we need to improve and generalize this mechanism.